### PR TITLE
dbeaver/pro#2711 Amazon Redshift: fix user and permission display

### PR DESCRIPTION
**Pull request by serjiokov**
Introduced API for item disable

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java [hunk 1]
```diff
@@ -479,15 +479,23 @@ private void injectTreeNodes(IConfigurationElement config) {
...
+            baseItem.clearChildren();
+            DBXTreeNode folderNode = baseItem.getParent();
+            folderNode.removeChild(baseItem);
+        }
+
         String changeFolderType = config.getAttribute("changeFolderType");
         if (changeFolderType != null) {
             DBXTreeNode folderNode = baseItem.getParent();
             if (folderNode instanceof DBXTreeFolder folder) {
+                System.out.println("changeFolderType " + changeFolderType);
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java]: !

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java (hunk 1 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java]: Already removed

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/meta/DBXTreeNode.java [hunk 2]
```diff
@@ -359,4 +359,14 @@ public void moveChildAfter(DBXTreeNode child, DBXTreeNode afterItem) {
         }
     }
 
+    /**
+     * Remove node item from the node list
+     *
+     * @param item - node for remove
+     */
+    public void disposeChild(DBXTreeItem item) {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/meta/DBXTreeNode.java]: disposeChild -> removeChild

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java [hunk 3]
```diff
@@ -51,6 +51,11 @@
  * DataSourceProviderDescriptor
  */
 public class DataSourceProviderDescriptor extends AbstractDescriptor implements DBPDataSourceProviderDescriptor {
+    private static final String ATTRIBUTE_CHANGE_FOLDER_LABEL = "changeFolderLabel"; //$NON-NLS-1$
+    private static final String ATTRIBUTE_CHANGE_FOLDER_TYPE = "changeFolderType"; //$NON-NLS-1$
+    private static final String ATTRIBUTE_DISPOSE = "dispose"; //$NON-NLS-1$
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java]: dispose->remove

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java (hunk 3 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java]: fixed

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/meta/DBXTreeNode.java (hunk 2 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/meta/DBXTreeNode.java]: fixed

**Pull request merged**